packages/react-core/src/components/Alert/__tests__/Alert.test.tsx

```diff
@@ -1,210 +1,724 @@
+  });
+  jest.useFakeTimers();
+
+  render(
+    <Alert title="Some title" timeout>
+      Some alert
+    </Alert>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.click(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  expect(screen.getByText('Some alert')).toBeVisible();
+  jest.useRealTimers();
+});
+
+test('Does not remove the alert on timeout if the user is hovered over the alert', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <Alert title="Some title" timeout>
+      Some alert
+    </Alert>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.hover(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  expect(alert).toBeVisible();
+  jest.useRealTimers();
+});
+
+test('Removes the alert after the user removes focus from the alert and 3000ms have passed', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <div>
+      <input />
+      <Alert title="Some title" timeout>
+        Some alert
+      </Alert>
+    </div>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.click(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  await user.click(screen.getByRole('textbox'));
+
+  act(() => {
+    jest.advanceTimersByTime(3000);
+  });
+
+  expect(screen.queryByText('Some alert')).not.toBeInTheDocument();
+  jest.useRealTimers();
+});
+
+test('Removes the alert after the user removes hover from the alert and 3000ms have passed', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <div>
+      <input />
+      <Alert title="Some title" timeout>
+        Some alert
+      </Alert>
+    </div>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.hover(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  await user.hover(screen.getByRole('textbox'));
+
+  act(() => {
+    jest.advanceTimersByTime(3000);
+  });
+
+  expect(screen.queryByText('Some alert')).not.toBeInTheDocument();
+  jest.useRealTimers();
+});
+
+test('Removes the alert after the user removes hover from the alert and timeoutAnimation time has passed', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <div>
+      <input />
+      <Alert title="Some title" timeout timeoutAnimation={1000}>
+        Some alert
+      </Alert>
+    </div>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.hover(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  await user.hover(screen.getByRole('textbox'));
```

Comment:
Didn't realize there was another similar test, but same question re: testing if the alert still exists after 8000ms and before moving hover.